go/apps/mixnet/queue.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2015, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixnet
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+)
+
+type sendQueueError struct {
+	id uint64 // Serial identifier of sender to which error pertains.
+	error
+}
+
+// The SendQueue structure.
+type SendQueue struct {
+	batchSize, ct int
+
+	network string
+
+	nextAddr   map[uint64]string
+	nextConn   map[uint64]net.Conn
+	sendBuffer map[uint64]*list.List
```

Comment:
Given that each sendBuffer value type is of a fixed length, why not use an array? Do you anticipate changing the buffer size in the course of operation?